File: cmd/root.go
package cmd

import (
	"os"

	"github.com/spf13/cobra"
)

// rootCmd represents the base command when called without any subcommands.
var rootCmd = &cobra.Command{
	Use:   "dynamic-node",
	Short: "dynamic-node build tool",
	Long:  "dynamic-node is a CLI driven by dynamic-node-cli.yaml for build/push/pull/clean and toolchain utilities for Node.js projects.",
}

// Execute adds all child commands to the root command and sets flags appropriately.
func Execute() {
	err := rootCmd.Execute()
	if err != nil {
		os.Exit(1)
	}
}

func init() {
}

File: src/cmd/root.js
import { Command } from "commander";
import { createBuildCommand } from "./build.js";
import { createCleanCommand } from "./clean.js";
import { createMetaCommand } from "./meta.js";
import { createPullCommand } from "./pull.js";
import { createPushCommand } from "./push.js";
import { createToolchainCommand } from "./toolchain.js";
import { createVersionCommand } from "./version.js";

export async function execute(argv = process.argv) {
  const rootCmd = new Command();

  rootCmd
    .name("dynamic-node")
    .description(
      "dynamic-node is a CLI driven by dynamic-node-cli.yaml for build/push/pull/clean and toolchain utilities for Node.js projects.",
    )
    .summary("dynamic-node build tool");

  rootCmd.addCommand(createBuildCommand());
  rootCmd.addCommand(createPushCommand());
  rootCmd.addCommand(createPullCommand());
  rootCmd.addCommand(createCleanCommand());
  rootCmd.addCommand(createMetaCommand());
  rootCmd.addCommand(createToolchainCommand());
  rootCmd.addCommand(createVersionCommand());

  await rootCmd.parseAsync(argv);
}
